dan_layer/engine_types/src/substate.rs

```diff
@@ -406,6 +421,13 @@ impl SubstateValue {
             _ => None,
         }
     }
+
+    pub fn into_transaction_receipt(&self) -> Option<&TransactionReceipt> {
```

Comment:
This is against rust convention. 

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

```suggestion
    pub fn into_transaction_receipt(self) -> Option<TransactionReceipt> {
```
and/or (as needed)

```suggestion
    pub fn as_transaction_receipt(&self) -> Option<&TransactionReceipt> {
```

and/or (as needed)

```suggestion
    pub fn as_transaction_receipt_mut(&mut self) -> Option<&mut TransactionReceipt> {
```